src/plugins/intel_gpu/tests/shape_infer/unsqueeze_si_test.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "test_utils.h"
+
+#include <intel_gpu/primitives/input_layout.hpp>
+#include <intel_gpu/primitives/reshape.hpp>
+#include <intel_gpu/primitives/data.hpp>
+
+#include "reshape_inst.h"
+
+#include "program_wrapper.h"
+
+#include <cmath>
+#include <algorithm>
+
+using namespace cldnn;
+using namespace ::tests;
+
+namespace shape_infer_tests {
+
+struct unsqueeze_test_params {
+    layout in_layout;
+    layout indices_layout;
+    std::vector<int64_t> indices_data;
+    ov::PartialShape output_partial_shape;
+    layout expected_layout;
+};
+
+class unsqueeze_test : public testing::TestWithParam<unsqueeze_test_params> { };
+TEST_P(unsqueeze_test, shape_infer) {
```

Comment:
I think squeeze/unsqueeze test cases should be a part of reshape_si_test file, as we don't have separate primitives for such operations.